pkg/ccl/streamingccl/logical/logical_replication_job.go

```diff
@@ -69,54 +71,40 @@ type logicalReplicationResumer struct {
 
 var _ jobs.Resumer = (*logicalReplicationResumer)(nil)
 
-// Resume is part of the jobs.Resumer interface.
-func (r *logicalReplicationResumer) Resume(ctx context.Context, execCtx interface{}) error {
-	jobExecCtx := execCtx.(sql.JobExecContext)
+func init() {
+	repstream.CreateRemoteProduceJobForLogicalReplicationHook = createRemoteProduceJobForLogicalReplication
+}
 
-	progress := r.job.Progress().Details.(*jobspb.Progress_LogicalReplication).LogicalReplication
-	payload := r.job.Details().(jobspb.LogicalReplicationDetails)
+func createRemoteProduceJobForLogicalReplication(
+	ctx context.Context, srcAddr string, tableNames []string,
+) (*streampb.ReplicationProducerSpec, error) {
+	// TODO(ssd): Copy over our GetFirstActiveClient logic
+	// so that we can connect to any node that we've seen
+	// in a previous Topology.
+	streamAddr, err := url.Parse(srcAddr)
+	if err != nil {
+		return nil, err
+	}
 
-	if len(progress.SourceSpans) == 0 {
-		// TODO(ssd): Move this out to the when we create the job.
-		//
-		// TODO(ssd): Copy over our GetFirstActiveClient logic
-		// so that we can connect to any node that we've seen
-		// in a previous Topology.
-		streamAddr, err := url.Parse(payload.TargetClusterConnStr)
-		if err != nil {
-			return err
-		}
+	client, err := streamclient.NewPartitionedStreamClient(ctx, streamAddr)
+	if err != nil {
+		return nil, err
+	}
+	defer func() { _ = client.Close(ctx) }()
```

Comment:
What's the reason behind wrapping this in an anyonymous function when you're not capturing the error output from client.Close()?
Wouldn't  
> defer client.Close(ctx) 

be just as good?

Reply:
linter complains about calling a function that returns an error and capturing its return value, to avoid accidentally forgetting to check an error, so he's capturing it (and discarding it) to make it clear/explicit that it is being discarded.